### PR TITLE
Mixed mode sanity

### LINKED ISSUE #11677: Revsquad appears in mixed mode
So I set up mixed to pick 3 modes at random
it picked revsquad
revs did not show up in tator menu
shuttle could not be called
r i p

### PR DESCRIPTION
**Pull request by gbasood**
All gamemodes now have a `can_be_mixed` var which defaults to 0
The modes which have this set to 1 are:
- Autotraitor
- Changeling
- Cult
- Vampire
- Wizard

Resolves #11677

### CODE REVIEW

File: code/game/gamemodes/mixed/mixed.dm [hunk 1]
```diff
@@ -56,29 +56,20 @@ var/global/list/mixed_allowed = list(
...
-																			)
+		var/list/datum/game_mode/possible = typesof(/datum/game_mode) - list(/datum/game_mode, datum/game_mode/mixed)
 		while(modes.len < 3)
 			if(!possible.len)
 				break
 			var/ourmode = pick(possible)
 			possible -= ourmode
 			var/datum/game_mode/M = new ourmode
+			// I put this in a separate block just in case BYOND does something silly with &&
+			if(!M.can_be_mixed)
```
Comment (ComicIronic) [inline code/game/gamemodes/mixed/mixed.dm]: Why create it? You can use initial() before the new.

**Comment by ComicIronic** [comment]
Use TRUE/FALSE

File: code/game/gamemodes/mixed/mixed.dm (hunk 1 above)
Comment (gbasood) [reply code/game/gamemodes/mixed/mixed.dm]: Give me the syntax and I'll use it then

Comment (ComicIronic) [reply code/game/gamemodes/mixed/mixed.dm]: You cast the typepath to the type you want and then do `initial(var.can_be_mixed)` on it.

**Comment by gbasood** [comment]
@ComicIronic TRUE/FALSE on var definitions is kinda dumb imo
Anyone messing with the code should realize what 0 and 1 evalute to in an if statement, so what's the point?

File: code/game/gamemodes/mixed/mixed.dm (hunk 1 above)
Comment (gbasood) [reply code/game/gamemodes/mixed/mixed.dm]: How do you typecast a child of game_mode here from a typesof list then

**Comment by Exxion** [comment]
There is no reason at all whatsoever not to use the defines

**Comment by ComicIronic** [comment]
But it's much clearer if I see your var and it's TRUE/FALSE and not 1/0. Is it always 1/0? Maybe there's a 2 out there somewhere.

Well, until we create a FILENOTFOUND define, anyways.

File: code/game/gamemodes/mixed/mixed.dm (hunk 1 above)
Comment (ComicIronic) [reply code/game/gamemodes/mixed/mixed.dm]: `var/datum/game_mode/ourmode = pick(possible)`

Comment (gbasood) [reply code/game/gamemodes/mixed/mixed.dm]: ~~literally doesn't work that way for me~~

**Comment by gbasood** [comment]
defines did

**Comment by clusterfack** [comment]
If 404 wasn't a magic number 9/11 could have been prevented

**Comment by clusterfack** [comment]
>302,796  1.01diff.txt


Wut

**Comment by gbasood** [comment]
fuck I thought I deleted that

**Pull request merged**